source/MRViewer/MRRibbonMenu.cpp

```diff
@@ -1690,10 +1690,16 @@ void RibbonMenu::drawSceneContextMenu_( const std::vector<std::shared_ptr<Object
             ImGui::EndTable();
         }
         ImGui::PopStyleVar();
-        //uncomment to close context menu on any change
-        //if ( wasChanged || wasAction )
-        if ( wasAction )
+
+
+        if ( ImGui::IsMouseDown( 2 ) && !( ImGui::IsAnyItemHovered() || ImGui::IsWindowHovered( ImGuiHoveredFlags_AnyWindow ) ) )
+        {
             ImGui::CloseCurrentPopup();
+        }
+        else if ( wasAction || ( wasChanged && closeOnAnyChange_ ) )
```

Comment:
why else if and not just if, this condotions are not connected

Reply:
so that the `ImGui::CloseCurrentPopup()` is not called twice